packages/core/streams/openai-stream.ts

```diff
@@ -1,27 +1,6 @@
-import {
-  AIStream,
-  trimStartOfStreamHelper,
-  type AIStreamCallbacks
-} from './ai-stream'
+import type { AIStreamOptions } from './types'
+import { AIStream } from './ai-stream'
 
-function parseOpenAIStream(): (data: string) => string | void {
-  const trimStartOfStream = trimStartOfStreamHelper()
-  return data => {
-    // TODO: Needs a type
-    const json = JSON.parse(data)
-
-    // this can be used for either chat or completion models
-    const text = trimStartOfStream(
-      json.choices[0]?.delta?.content ?? json.choices[0]?.text ?? ''
-    )
-
-    return text
-  }
-}
-
-export function OpenAIStream(
-  res: Response,
-  cb?: AIStreamCallbacks
-): ReadableStream {
-  return AIStream(res, parseOpenAIStream(), cb)
+export function OpenAIStream(res: Response, options: AIStreamOptions = {}) {
+  return AIStream(res, { platform: 'openai', ...options })
```

Comment:
Here's how platform implementations are set up under the new pattern: A platform is added to the parsers dictionary and `AIStream` will look it up. This lets us manage all of the parsers in one place (`parser.ts`), which for now only contains text parsers, but soon will probably contain others (such as JSON parsers).